src/core/computedvalue.ts

```diff
@@ -20,18 +19,21 @@ export interface IComputedValue<T> {
  * If a computed value isn't actively used by another observer, but is inspect, it will compute lazily to return at least a consistent value.
  */
 export class ComputedValue<T> implements IObservable, IComputedValue<T>, IDerivation {
-	isLazy = true; // nobody is observing this derived value, so don't bother tracking upstream values
 	isComputing = false;
-	staleObservers: IDerivation[] = [];
-	observers = new SimpleSet<IDerivation>();      // nodes that are dependent on this node. Will be notified when our state change
+
+	dependenciesState = -1;
 	observing = [];       // nodes we are looking at. Our value depends on these nodes
+	newObserving = null; // during tracking it's array with new observed observers
+
+	isPendingUnobservation: boolean; // for effective unobserving
+	isObserved = false;
```

Comment:
the following invariant if I'm correct? `isObserved === dependenciesState !== -1`? That might indicate that `isObserved` might not be explicitly needed?